plugins/com.gwtplugins.gdt.eclipse.suite/src/com/google/gdt/eclipse/suite/wizards/WebAppTemplateProjectCreator.java

```diff
@@ -0,0 +1,435 @@
+
+      /*
+       * Refresh contents; if this project was generated via GWT's WebAppCreator, then these files would have been created
+       * directly on the file system Although, this refresh should have been done via the project.open() call, which is
+       * part of the createProject call above.
+       */
+      project[i].refreshLocal(IResource.DEPTH_INFINITE, monitor);
+
+      // Set all of the natures on the project
+      NatureUtils.addNatures(project[i], projectTemplate.getNatureIds(i));
+
+      // Create the java project
+      createdJavaProjects[i] = JavaCore.create(project[i]);
+
+      // Create a source folder and add it to the raw classpath
+      IResource warFolder = project[i].findMember(WebAppUtilities.DEFAULT_WAR_DIR_NAME);
+      boolean createWarFolders = (warFolder != null);
+      createFolders(project[i], createWarFolders, monitor);
+      if (createWarFolders) {
+        // Set the WAR source/output directory to "/war"
+        WebAppUtilities.setDefaultWarSettings(project[i]);
+
+        // Set the default output directory
+        WebAppUtilities.setOutputLocationToWebInfClasses(createdJavaProjects[i], monitor);
+        /**
+         * Copy files into the web-inf lib folder. This code assumes that it is running in a context that has a workspace
+         * lock.
+         */
+        Sdk gwtSdk = getGWTSdk();
+        setGwtSdk(gwtSdk);
+        if (gwtSdk != null) {
+          new GWTUpdateWebInfFolderCommand(createdJavaProjects[i], gwtSdk).execute();
+        }
+      }
+      reformatJavaFiles(dir);
+
+      // Allow other extensions to run after project creation
+      includeExtensionPartipants();
+    }
+  }
+
+  /**
+   * Replaces all occurrences of the PACKAGENAME with the given package name.
+   * @param iProject
+   * @param dir
+   * @throws IOException
+   */
+  private void replacePackageNames(IProject iProject, File dir) throws IOException {
+    Stack<File> files = new Stack<>();
+    files.add(dir);
+    while(files.isEmpty() == false)
+    {
+      File file = files.pop();
+      if(file.getName().contains("_PACKAGENAME_"))
+      {
+        String name = file.getName();
+        name = name.replace("_PACKAGENAME_", packageBaseName.replace('.', File.separatorChar));
+        File newFile = new File(file.getParentFile(), name);
+        newFile.getParentFile().mkdirs();
+        file.renameTo(newFile);
+        file = newFile;
+      }
+      if(file.getName().contains("_.._"))
+      {
+        String name = file.getName();
+        name = name.replace("_.._", "../");
+        File newFile = new File(file.getParentFile(), name);
+        newFile.getParentFile().mkdirs();
+        file.renameTo(newFile);
+        file = newFile;
+      }
+      if(file.isDirectory())
+      {
+        File[] cf = file.listFiles();
+        files.addAll(Arrays.asList(cf));
+      }
+      else
+      {
+        replaceInFile(file.toString(), "\\$\\{PACKAGENAME\\}", packageBaseName);
```

Comment:
I'm not a huge fan of supporting multiple template patterns in the same file, but maybe we can be consistent here, and always use `_` for paths, and `${}`s for strings?

That said, replaceInFile never actually accepts a pattern - why not change the string to take a literal, and use `String pattern = Pattern.quote(findMe)` to rewrite? then we never need to worry about mixing tokens